server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBranchesMenuItem.tsx

```diff
@@ -20,48 +20,125 @@
 import * as React from 'react';
 import { Link } from 'react-router';
 import * as classNames from 'classnames';
+import DeleteBranchModal from './DeleteBranchModal';
+import RenameBranchModal from './RenameBranchModal';
 import BranchStatus from '../../../../components/common/BranchStatus';
 import { Branch, Component } from '../../../types';
 import BranchIcon from '../../../../components/icons-components/BranchIcon';
+import ChangeIcon from '../../../../components/icons-components/ChangeIcon';
+import DeleteIcon from '../../../../components/icons-components/DeleteIcon';
 import { isShortLivingBranch } from '../../../../helpers/branches';
 import { translate } from '../../../../helpers/l10n';
 import { getProjectBranchUrl } from '../../../../helpers/urls';
 
-interface Props {
+export interface Props {
   branch: Branch;
+  canAdmin?: boolean;
   component: Component;
+  onBranchesChange: () => void;
   onSelect: (branch: Branch) => void;
   selected: boolean;
 }
 
-export default function ComponentNavBranchesMenuItem({ branch, ...props }: Props) {
-  const handleMouseEnter = () => {
-    props.onSelect(branch);
+interface State {
+  deleteBranchModal: boolean;
+  renameBranchModal: boolean;
+}
+
+export default class ComponentNavBranchesMenuItem extends React.PureComponent<Props, State> {
+  state: State = { deleteBranchModal: false, renameBranchModal: false };
+
+  handleMouseEnter = () => {
+    this.props.onSelect(this.props.branch);
+  };
+
+  handleDeleteBranchClick = (event: React.SyntheticEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    event.currentTarget.blur();
+    this.setState({ deleteBranchModal: true });
+  };
+
+  handleDeleteBranchClose = () => {
+    this.setState({ deleteBranchModal: false });
```

Comment:
I'm not sure where the problem is, but when clicking *cancel* on one of the modal the branch drop down is also closed.